server/src/test/java/io/crate/lucene/CommonQueryBuilderTest.java

```diff
@@ -615,26 +615,6 @@ public void test_eq_on_alias_uses_termquery() throws Exception {
         assertThat(query).isExactlyInstanceOf(TermQuery.class);
     }
 
-    @Test
-    public void test_eq_on_alias_inner_func_uses_termquery() throws Exception {
```

Comment:
- `eq` operator simply returns `null` if the first arg symbol is not a ref: https://github.com/crate/crate/blob/60c6d5956ab1b7302e1a9400c3bfdad26d72fa71/server/src/main/java/io/crate/expression/operator/EqOperator.java#L113-L115
so the behaviour of `eq_on_alias_inner_func` solely depend on the inner func.

- the test asserts for `BooleanQuery` not `TermQuery` as opposed to the test name.
- if I change `array_length` to `array_lower` it becomes `GenericFunctionQuery` also not as suggested by the test name.

Reply:
afaik this test was introduced when we fixed a performance regression. 

Can you please check that it's not degraded?

Concrete scenario (corresponding to this simple test) can be found in https://github.com/crate/crate/issues/11846#issuecomment-949691989 (see issue description to get schema and also spec file to test this)

Reply:
The two tests added by #11846 seem to be passing with https://github.com/crate/crate/pull/11854/files#diff-36c2ee48849d03b2e21d18116b8e96b1128521516442f02ae7ea707be4640806R147 commented out which should be disabling the entire changeset.
I think we can disable them and add a todo entry for feature freeze?

Reply:
Please check with https://github.com/crate/crate-benchmarks/blob/21791e2f6227d4d3abb6fc67e039a2425f7a7705/specs/select/aliased_object_key.toml that we don't have a regression described in https://github.com/crate/crate/issues/11846 again. Otherwise LGTM

Reply:
It looks like below scenario also captures the bug:
```
cr> create table t (a text storage with (columnstore = 'false'));                                                                                                     
CREATE OK, 1 row affected  (0.365 sec)
cr> insert into t select * from generate_series(1,1000000);                                                                                                           
INSERT OK, 1000000 rows affected  (10.321 sec)
cr> create view v as select a as a2 from t;                                                                                                                           
CREATE OK, 1 row affected  (0.199 sec)
cr> create view v2 as select a2 as a3 from v;                                                                                                                         
CREATE OK, 1 row affected  (0.031 sec)
cr> create view v3 as select a3 as a4 from v2;                                                                                                                        
CREATE OK, 1 row affected  (0.025 sec)
cr> create view v4 as select a4 as a5 from v3;                                                                                                                        
CREATE OK, 1 row affected  (0.024 sec)
cr> select count(*) from t where a = '34534534' or a = '4456';                                                                                                        
+----------+
| count(*) |
+----------+
|        1 |
+----------+
SELECT 1 row in set (0.071 sec)
cr> select count(*) from v4 where a5 = '34534534' or a5 = '4456';                                                                                                     
+----------+
| count(*) |
+----------+
|        1 |
+----------+
SELECT 1 row in set (16.890 sec) --> very slow
```
This fails on `4.6.4` fixed for `4.6.5` and still good on `master`. Also, disabling `AliasResolver` on `master` causes the perf regressions:
```
cr> select count(*) from t where a = '34534534' or a = '4456';                                                                                                        
+----------+
| count(*) |
+----------+
|        2 |
+----------+
SELECT 1 row in set (2.881 sec)
cr> select count(*) from v4 where a5 = '34534534' or a5 = '4456';                                                                                                     
+----------+
| count(*) |
+----------+
|        2 |
+----------+
SELECT 1 row in set (34.984 sec)
```
I will continue on this separately.

Reply:
> Also, disabling AliasResolver(equivalent to reverting the fix) on master causes the perf regressions

This is not correct, see https://github.com/crate/crate/pull/16406 for updates.